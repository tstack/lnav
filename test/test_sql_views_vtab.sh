#! /bin/bash

export TZ=UTC
export YES_COLOR=1
unset XDG_CONFIG_HOME

run_test ${lnav_test} -n \
    -c ";SELECT view_name,basename(filepath),visible FROM lnav_view_files" \
    -c ":write-csv-to -" \
    ${test_dir}/logfile_access_log.*

check_output "lnav_view_files does not work?" <<EOF
view_name,basename(filepath),visible
log,logfile_access_log.0,1
log,logfile_access_log.1,1
EOF

run_cap_test ${lnav_test} -n \
    -c ";UPDATE lnav_view_files SET visible=0 WHERE endswith(filepath, 'log.0')" \
    ${test_dir}/logfile_access_log.*

run_test ${lnav_test} -n \
    -c ";DELETE FROM lnav_view_stack" \
    ${test_dir}/logfile_access_log.0

check_output "deleting the view stack does not work?" <<EOF
EOF

run_cap_test ${lnav_test} -n \
    -c ";UPDATE lnav_view_stack SET name = 'foo'" \
    ${test_dir}/logfile_access_log.0

run_cap_test ${lnav_test} -n \
    -c ";INSERT INTO lnav_view_stack VALUES ('help')" \
    -c ";DELETE FROM lnav_view_stack WHERE name = 'log'" \
    ${test_dir}/logfile_access_log.0

run_cap_test ${lnav_test} -n \
    -c ";INSERT INTO lnav_view_filters VALUES ('log', 0, 1, 'out', 'regex', '')" \
    ${test_dir}/logfile_access_log.0

run_cap_test ${lnav_test} -n \
    -c ";INSERT INTO lnav_view_filters VALUES ('log', 0, 1, 'out', 'regex', 'abc(')" \
    ${test_dir}/logfile_access_log.0

run_cap_test ${lnav_test} -n \
    -c ";INSERT INTO lnav_view_filters VALUES ('bad', 0, 1, 'out', 'regex', 'abc')" \
    ${test_dir}/logfile_access_log.0

run_cap_test ${lnav_test} -n \
    -c ";INSERT INTO lnav_view_filters VALUES (NULL, 0, 1, 'out', 'regex', 'abc')" \
    ${test_dir}/logfile_access_log.0

run_cap_test ${lnav_test} -n \
    -c ";INSERT INTO lnav_view_filters VALUES ('log', 0 , 1, 'bad', 'regex', 'abc')" \
    ${test_dir}/logfile_access_log.0

run_cap_test ${lnav_test} -n \
    -c ";INSERT INTO lnav_view_filters (view_name, pattern) VALUES ('log', 'vmk')" \
    ${test_dir}/logfile_access_log.0

run_cap_test ${lnav_test} -n \
    -c ";INSERT INTO lnav_view_filters (view_name, pattern, type) VALUES ('log', 'vmk', 'in')" \
    ${test_dir}/logfile_access_log.0

run_cap_test ${lnav_test} -n \
    -c ";INSERT INTO lnav_view_filters (view_name, pattern, type) VALUES ('log', 'vmk', 'in')" \
    -c ";INSERT INTO lnav_view_filters (view_name, pattern, type) VALUES ('log', 'vmk', 'in')" \
    -c ';SELECT * FROM lnav_view_filters' \
    -c ':write-screen-to -' \
    ${test_dir}/logfile_access_log.0

run_cap_test ${lnav_test} -n \
    -c ";INSERT INTO lnav_view_filters (view_name, pattern, type) VALUES ('log', 'vmk', 'in')" \
    -c ";INSERT INTO lnav_view_filters (view_name, pattern, type) VALUES ('log', 'vmk1', 'in')" \
    -c ";UPDATE lnav_view_filters SET pattern = 'vmk'" \
    -c ';SELECT * FROM lnav_view_filters' \
    -c ':write-screen-to -' \
    ${test_dir}/logfile_access_log.0

run_cap_test ${lnav_test} -n \
    -c ";INSERT INTO lnav_view_filters (view_name, language, pattern) VALUES ('log', 'sql', '1')" \
    -c ";INSERT INTO lnav_view_filters (view_name, language, pattern) VALUES ('log', 'sql', '1')" \
    -c ';SELECT * FROM lnav_view_filters' \
    -c ':write-screen-to -' \
    ${test_dir}/logfile_access_log.0

run_cap_test ${lnav_test} -n \
    -c ":filter-out vmk" \
    -c ";DELETE FROM lnav_view_filters" \
    ${test_dir}/logfile_access_log.0

run_cap_test ${lnav_test} -n \
    -c ":filter-out vmk" \
    -c ";UPDATE lnav_view_filters SET pattern = 'vmkboot'" \
    ${test_dir}/logfile_access_log.0

run_test ${lnav_test} -n \
    -c ":filter-out vmk" \
    -c ";SELECT * FROM lnav_view_filter_stats" \
    -c ":write-csv-to -" \
    ${test_dir}/logfile_access_log.0

check_output "view filter stats is not working?" <<EOF
view_name,filter_id,hits
log,1,2
EOF

run_test ${lnav_test} -n \
    -c ";INSERT INTO lnav_view_filters (view_name, language, pattern) VALUES ('log', 'sql', ':sc_bytes = 134')" \
    ${test_dir}/logfile_access_log.0

check_output "inserted filter-out did not work?" <<EOF
192.168.202.254 - - [20/Jul/2009:22:59:26 +0000] "GET /vmw/cgi/tramp HTTP/1.0" 200 134 "-" "gPXE/0.9.7"
EOF

run_test ${lnav_test} -n \
    -c ';DELETE FROM lnav_views' \
    -c ';SELECT count(*) FROM lnav_views' \
    -c ':write-csv-to -' \
    ${test_dir}/logfile_access_log.0

check_output "delete from lnav_views table works?" <<EOF
count(*)
9
EOF


run_test ${lnav_test} -n \
    -c ";INSERT INTO lnav_views (name) VALUES ('foo')" \
    -c ';SELECT count(*) FROM lnav_views' \
    -c ':write-csv-to -' \
    ${test_dir}/logfile_access_log.0

check_output "insert into lnav_views table works?" <<EOF
count(*)
9
EOF

run_cap_test ${lnav_test} -n \
    -c ";UPDATE lnav_views SET top = 1 WHERE name = 'log'" \
    ${test_dir}/logfile_access_log.0

run_test ${lnav_test} -n \
    -c ";UPDATE lnav_views SET top = inner_height - 1 WHERE name = 'log'" \
    ${test_dir}/logfile_access_log.0

check_output "updating lnav_views.top using inner_height does not work?" <<EOF
192.168.202.254 - - [20/Jul/2009:22:59:29 +0000] "GET /vmw/vSphere/default/vmkernel.gz HTTP/1.0" 200 78929 "-" "gPXE/0.9.7"
EOF


run_cap_test ${lnav_test} -n \
    -c ";UPDATE lnav_views SET top_time = 'bad-time' WHERE name = 'log'" \
    ${test_dir}/logfile_access_log.0

run_cap_test ${lnav_test} -n \
    -c ";UPDATE lnav_views SET top_time = '2014-10-08T00:00:00' WHERE name = 'log'" \
    ${test_dir}/logfile_generic.0

run_cap_test ${lnav_test} -n \
    -c ";UPDATE lnav_views SET search = 'warn' WHERE name = 'log'" \
    -c ";SELECT search FROM lnav_views WHERE name = 'log'" \
    ${test_dir}/logfile_generic.0

run_cap_test ${lnav_test} -n \
    -c ";UPDATE lnav_views SET search = 'warn' WHERE name = 'log'" \
    -c ":goto 0" \
    -c ":next-mark search" \
    ${test_dir}/logfile_generic.0

run_cap_test ${lnav_test} -n \
    -c ";UPDATE lnav_views SET top_meta = json_object('anchor', '#build') WHERE name = 'text'" \
    ${top_srcdir}/README.md

run_cap_test ${lnav_test} -n \
    -c ":goto 5" \
    -c ";SELECT top_meta FROM lnav_top_view" \
    -c ":write-json-to -" \
    ${test_dir}/logfile_xml_msg.0

run_cap_test ${lnav_test} -n -I ${test_dir} \
    -c ";UPDATE lnav_views SET options = json_object('row-details', 'show') WHERE name = 'log'" \
    -c ":goto 2" \
    ${test_dir}/logfile_xml_msg.0

run_cap_test ${lnav_test} -n -I ${test_dir} \
    -c ";UPDATE lnav_views SET options = json_object('row-details', 'show') WHERE name = 'log'" \
    -c ":goto 9" \
    ${test_dir}/logfile_bunyan.0

run_cap_test ${lnav_test} -n \
    -c ";UPDATE lnav_views SET top_meta = json_object('file', 'bad') WHERE name = 'text'" \
    ${test_dir}/textfile_ansi.0
