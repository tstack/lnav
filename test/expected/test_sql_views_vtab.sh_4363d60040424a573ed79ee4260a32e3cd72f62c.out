2023-03-24T14:26:16.243 renovate[7] INFO     Dependency extraction complete
 Received Time: [1m2023-03-24T14:26:16.243[0m â€” [1min the future[0m  Format: [1m%Y-%m-%dT%H:%M:%S.%L%z[0m
 Known message fields for table [1mbunyan_log[0m:
   name     = [1mrenovate[0m
   hostname = [1mrenovate-gitlab-67c4bcb5-9ggbv[0m
   pid      = [1m7[0m
   level    = [1m30[0m
   v        = [1m0[0m
 JSON fields:
   [1mjget[0m(log_raw_text, [35m'/baseBranch'[0m) = [1mmain[0m
   [1mjget[0m(log_raw_text, [35m'/logContext'[0m) = [1mqjifsaDDI[0m
   [1mjget[0m(log_raw_text, [35m'/repository'[0m) = [1mwebgui/custom-icons-transformer[0m
   [1mjget[0m(log_raw_text, [35m'/stats'[0m) = [1m{"managers":{"gitlabci":{"fileCount":1,"depCount":1},"gitlabci-include":{"fileCount":1,"depCount":1},"npm":{"fileCount":1,"depCount":15}},"total":{"fileCount":3,"depCount":17}}[0m
 No discovered message fields
  logContext: qjifsaDDI
  repository: webgui/custom-icons-transformer
  baseBranch: main
  stats: {"managers":{"gitlabci":{"fileCount":1,"depCount":1},"gitlabci-include":{"fileCount":1,"depCount":1},"npm":{"fileCount":1,"depCount":15}},"total":{"fileCount":3,"depCount":17}}
2023-03-24T14:26:16.390 renovate[7] DEBUG    Dependency node has unsupported/unversioned value lts-bullseye-slim (versioning=docker)
  logContext: qjifsaDDI
  repository: webgui/custom-icons-transformer
2023-03-24T14:26:17.493 renovate[7] DEBUG    Release 2.8.7 is pending status checks
  logContext: qjifsaDDI
  repository: webgui/custom-icons-transformer
  depName: prettier
  check: stabilityDays
2023-03-24T14:26:17.897 renovate[7] DEBUG    Release 4.4.1 is pending status checks
  logContext: qjifsaDDI
  repository: webgui/custom-icons-transformer
  depName: rimraf
  check: stabilityDays
2023-03-24T14:26:17.897 renovate[7] DEBUG    All releases are pending - using latest
  logContext: qjifsaDDI
  repository: webgui/custom-icons-transformer
  depName: rimraf
  bucket: non-major
2023-03-24T14:26:18.330 renovate[7] DEBUG    Release 2.10.0 is pending status checks
  logContext: qjifsaDDI
  repository: webgui/custom-icons-transformer
  depName: prettier-plugin-svelte
  check: stabilityDays
2023-03-24T14:26:18.331 renovate[7] DEBUG    All releases are pending - using latest
  logContext: qjifsaDDI
  repository: webgui/custom-icons-transformer
  depName: prettier-plugin-svelte
  bucket: non-major
